staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress_test.go

```diff
@@ -0,0 +1,142 @@
+	"k8s.io/kubectl/pkg/scheme"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func TestCreateIngressBinding(t *testing.T) {
+
+	expectBinding := &v1beta1.Ingress{
+		TypeMeta: v1.TypeMeta{
+			APIVersion: "extensions/v1beta1",
+			Kind:       "Ingress",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name: "fake-ingress",
+		},
+		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				v1beta1.IngressRule{
+					Host: "foo.bar.com",
+					IngressRuleValue: v1beta1.IngressRuleValue{
+						HTTP: &v1beta1.HTTPIngressRuleValue{
+							Paths: []v1beta1.HTTPIngressPath{
+								v1beta1.HTTPIngressPath{
+									Path: "/api",
+									Backend: v1beta1.IngressBackend{
+										ServiceName: "fake-service",
+										ServicePort: intstr.FromInt(8080),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	tf := cmdtesting.NewTestFactory().WithNamespace("test")
+	defer tf.Cleanup()
+
+	ns := scheme.Codecs
+
+	info, _ := runtime.SerializerInfoForMediaType(ns.SupportedMediaTypes(), runtime.ContentTypeJSON)
+	encoder := ns.EncoderForVersion(info.Serializer, schema.GroupVersion{Group: "extensions", Version: "v1beta1"})
+	decoder := ns.DecoderToVersion(info.Serializer, schema.GroupVersion{Group: "extensions", Version: "v1beta1"})
+
+	tf.Client = &IngressRESTClient{
+		RESTClient: &fake.RESTClient{
+			NegotiatedSerializer: ns,
+			Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+				switch p, m := req.URL.Path, req.Method; {
+				case p == "/namespaces/test/rolebindings" && m == "POST":
+					bodyBits, err := ioutil.ReadAll(req.Body)
+					if err != nil {
+						t.Fatalf("TestCreateIngressBinding error: %v", err)
+						return nil, nil
+					}
+
+					if obj, _, err := decoder.Decode(bodyBits, nil, &v1beta1.Ingress{}); err == nil {
+						if !reflect.DeepEqual(obj.(*v1beta1.Ingress), expectBinding) {
+							t.Fatalf("TestCreateIngressBinding: expected:\n%#v\nsaw:\n%#v", expectBinding, obj.(*v1beta1.Ingress))
+							return nil, nil
+						}
+					} else {
+						t.Fatalf("TestCreateIngressBinding error, could not decode the request body into v1beta1.Ingress object: %v", err)
+						return nil, nil
+					}
+
+					responseBinding := &v1beta1.Ingress{}
+					responseBinding.Name = "fake-ingress"
+					return &http.Response{StatusCode: 201, Header: cmdtesting.DefaultHeader(), Body: ioutil.NopCloser(bytes.NewReader([]byte(runtime.EncodeOrDie(encoder, responseBinding))))}, nil
+				default:
+					t.Fatalf("unexpected request: %#v\n%#v", req.URL, req)
+					return nil, nil
+				}
+			}),
+		},
+	}
+
+	cmd := NewCmdCreateIngress(tf, genericclioptions.NewTestIOStreamsDiscard())
+	cmd.Flags().Set("host", "foo.bar.com")
+	cmd.Flags().Set("path", "/api")
+	cmd.Flags().Set("service-name", "fake-service")
+	cmd.Flags().Set("service-port", "8080")
+	cmd.Run(cmd, []string{"fake-ingress"})
+}
+
+type IngressRESTClient struct {
+	*fake.RESTClient
+}
+
+func (c *IngressRESTClient) Post() *restclient.Request {
```

Comment:
That seems overly complicated. Have a look at other create command and their testing suites.

Reply:
Looked at other commands as you suggested and got this sorted out